sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -717,27 +706,26 @@ public synchronized void enqueue(@NonNull Log log, @NonNull final String groupNa
     /**
      * Check for logs to trigger immediately or schedule with a timer or does nothing if no logs.
      *
-     * @param groupName the group name.
+     * @param groupState the group state.
      */
     @VisibleForTesting
-    synchronized void checkPendingLogs(@NonNull String groupName) {
-        GroupState groupState = mGroupStates.get(groupName);
+    synchronized void checkPendingLogs(@NonNull GroupState groupState) {
         if (groupState.mPaused) {
-            AppCenterLog.debug(LOG_TAG, groupName + " is paused. Skip checking pending logs.");
+            AppCenterLog.debug(LOG_TAG, groupState.mName + " is paused. Skip checking pending logs.");
             return;
         }
         long pendingLogCount = groupState.mPendingLogCount;
-        AppCenterLog.debug(LOG_TAG, "checkPendingLogs(" + groupName + ") pendingLogCount=" + pendingLogCount);
+        AppCenterLog.debug(LOG_TAG, "checkPendingLogs(" + groupState.mName + ") pendingLogCount=" + pendingLogCount);
         if (pendingLogCount >= groupState.mMaxLogsPerBatch) {
-            triggerIngestion(groupName);
+            triggerIngestion(groupState);
         } else if (pendingLogCount > 0 && !groupState.mScheduled) {
             groupState.mScheduled = true;
             mAppCenterHandler.postDelayed(groupState.mRunnable, groupState.mBatchTimeInterval);
         }
     }
 
     @VisibleForTesting
-    GroupState getGroupState(String groupName) {
+    GroupState getGroupState(@SuppressWarnings("SameParameterValue") String groupName) {
```

Comment:
Not a big deal but we used to have method level suppress warning in this case. Can we change?

Reply:
This one is parameter specific though. We have other examples of this one in the repo.

Reply:
I thought this is a kind of rule what we do but you are saying it isn't. OK. I'd prefer this way since this will hide some warnings that we don't expect in other parameters. 
There is an example that we are not doing this one. We probably need to check all of these cases in the future (low priority).
[Example](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java#L159)

Reply:
Yes this should have been reviewed as suppressing warnings for all parameters does not help knowing what parameter is always used with 1 value.

One other exception is trycanusefinally that can be at class level as this is more global as we can't use it because of min API level so it does not make sense to put it in each method or statement.